### DISCUSSION TITLE
helix defaults to 2 spaces for indentation

### DISCUSSION THREAD

**Discussion by soupfoo** [helix-editor/helix#8013] [Q&A] [answered]
How to change tab size to 4 spaces ? Helix seems to default to 2 spaces for indentation.
I don't want to run `:indent-style 4` every time I open the editor. Is there any parameter to put in `~/.config/helix/config.toml` ?

**Comment by soupfoo** [comment]
It does follow the indentation properly once I save the file with the desired indentation.

**Comment by erasin** [comment, marked as answer]
Helix will detect the indentation used in the file, or otherwise defaults to the language indentation configured in [`languages.toml`](https://github.com/helix-editor/helix/blob/master/languages.toml), with a fallback to tabs if it isn't specified. Line ending is likewise auto-detected, and will remain unchanged if no line endings were detected.

Set Config with `~/.config/helix/languages.toml` , like config of javascript language 

```
[[language]]
name = "javascript"
indent = { tab-width = 4, unit = "    " }
```

**Comment by soupfoo** [comment]
ohh ! 😮
thank you !

**Comment by haasn** [comment]
How do I change the default indentation style for empty files? `:lang` just reports `text`. I tried adding it to `languages.toml` but it doesn't work:

```toml
[[language]]
name = "text"
indent = { tab-width = 4, unit = "    " }
```

```
Failed to parse language config: missing field `scope`
in `language`

Press <ENTER> to continue with default language config
```

**Comment by Axlefublr** [comment]
yeah I want to set the tab width to 3 and unit to a tab, to act as my default for files that are text. trying to set the configuration for text doesn't work

**Comment by ghost** [comment]
Searching on how to set my tab width, I found this discussion and would like to add that using the line<br> `indent = { tab-width = 4, unit = "    " }` <br>without setting a `[[language]]` section seems to have set it globally, which is what I had hoped for.

**Comment by jakobjpeters** [comment]
@nieuemma Is this placed at the top of `languages.toml`? It's not working for me.

**Reply by ghost** [reply]
Trying it now it doesn't seem to work the way I did it before. Not sure if I was imagining it or not. Apologies.

**Reply by jakobjpeters** [reply]
Thank you for checking!
